File: src/GamesFinder.Orchestrator.Domain/Classes/Entities/Game.cs
using GamesFinder.Orchestrator.Domain.Enums;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace GamesFinder.Orchestrator.Domain.Classes.Entities;

public class Game(
  string name,
  int steamID,
  List<GameOffer>? initialOffers = null,
  string? description = null,
  string? steamUrl = null,
  string? headerImage = null
    ) : Entity
{
  [BsonElement("name")]
  public string Name { get; set; } = name;

  [BsonElement("steam_url")]
  public string? SteamURL { get; set; } = steamUrl;
  [BsonElement("steam_id")]
  public int SteamID { get; set; } = steamID;
  [BsonElement("in_packages")]
	public List<int> InPackages { get; set; } = new();
	[BsonElement("isDLC")]
	public bool IsDLC { get; set; }
  [BsonElement("description")]
  public string? Description { get; set; } = description;
  [BsonElement("header_image")]
  public string? HeaderImage { get; set; } = headerImage;
  [BsonIgnore]
  public List<GameOffer> Offers = initialOffers ?? new List<GameOffer>();
  [BsonElement("is_released")]
  public bool IsReleased { get; set; }
  [BsonElement("initial_prices")]
  public Dictionary<ECurrency, decimal> InitialPrices = new();

}

File: workers/src/utils/types/entities/game.ts
import { eCurrency } from "../enums/eCurrency.js";
import { Entity } from "./entity.js";
import { GameOffer } from "./gameOffer.js";

export interface Game extends Entity {
  name: string;
  steamUrl: string;
  steamID: number;
  inPackages: number[] | null;
  isDLC: boolean;
  description: string | null;
  headerImage: string | null;
  offers: GameOffer[] | null;
  isReleased: boolean;
  initialPrices: Record<eCurrency, number> | null;
}